src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1501,22 +1514,44 @@ internal void UpdateModelValue(IEnumerable<Guid> modelGuids, string propertyName
 
         internal string GetStringRepOfWorkspace()
         {
-            // Create the xml document to write to.
-            var document = new XmlDocument();
-            document.CreateXmlDeclaration("1.0", null, null);
-            document.AppendChild(document.CreateElement("Workspace"));
+            try
+            {
+                //Send the data in JSON format
+                //step 1: convert the model to json
+                var json = this.ToJson(null);
 
-            //This is only used for computing relative offsets, it's not actually created
-            string virtualFileName = Path.Combine(Path.GetTempPath(), "DynamoTemp.dyn");
-            Utils.SetDocumentXmlPath(document, virtualFileName);
+                //step2 : parse it as JObject
+                var jo = JObject.Parse(json);
 
-            if (!PopulateXmlDocument(document))
-                return String.Empty;
+                //step 2: raise the event to populate the view block
+                if (PopulateJSONWorkspace != null)
+                {
+                    json = PopulateJSONWorkspace(jo);
+                }
 
-            //Now unset the temp file name again
-            Utils.SetDocumentXmlPath(document, null);
+                return json;
+            }
+            catch (JsonReaderException ex)
+            {
+                JArray array = new JArray();
+                array.Add(ex.Message);
+
+                JObject jo = new JObject();
+                jo["exception"] = array;
+
+                return jo.ToString();
```

Comment:
so just so I am clear, this returns valid json?

Reply:
yes but this is like
```
{
exception: exception
}
```